### PR TITLE
parser

### PR DESCRIPTION
**Pull request by devlordduck**
idk what to put here

### CODE REVIEW

**Comment by devlordduck** [comment]
DAMN I LOST THE PARSERS.JS

**Comment by devlordduck** [comment]
nvm found it

**Comment by devlordduck** [comment]
uh

File: src/handler/parsers.js [hunk 1]
```diff
@@ -522,42 +49,778 @@ const SlashOptionsParser = async (options) => {
...
+ * @returns {boolean} - The result.
+*/
+let checkBoolean = (text, orelse) => {
+    if (orelse && typeof orelse !== "boolean") return;
+    if (!text  || typeof text   !== "string" ) return orelse;
+
+    if ( text && text.trim() && (text.trim() === "" || text.trim() === " ") ) 
+        return orelse;
+
+    return text?.trim()?.toLowerCase() === "true" ? true : (text?.trim()?.toLowerCase() === "false" ? false : orelse);
```
Comment (Faf4a) [inline src/handler/parsers.js:66]: Can be replaced with Boolean() I assume? No need for that as yes/no support is gone.

File: src/handler/parsers.js [hunk 2]
```diff
@@ -522,42 +49,778 @@ const SlashOptionsParser = async (options) => {
...
+                            };
+            
+                            id = id.replace(">", "");
+                            emoji = {
+                                name: name,
+                                id: id === "" ? undefined : id,
+                                animated: animated.replace("<", "") === "a"
+                            };
+                        };
+
```
Comment (Faf4a) [inline src/handler/parsers.js:288]: Using `getEmoji()` would allow more possibilities.

File: src/handler/parsers.js [hunk 3]
```diff
@@ -522,42 +49,778 @@ const SlashOptionsParser = async (options) => {
...
+                                    animated = "";
+                                };
+                
+                                id = id.replace(">", "");
+                                emoji = {
+                                    name: name,
+                                    id: id === "" ? undefined : id,
+                                    animated: animated.replace("<", "") === "a"
+                                };
+                            };
```
Comment (Faf4a) [inline src/handler/parsers.js:352]: Using `getEmoji()` would allow more possibilities here as well.

File: src/handler/parsers.js [hunk 4]
```diff
@@ -522,42 +49,778 @@ const SlashOptionsParser = async (options) => {
...
+            const trueorfalse = "true";
+
+            if (Checker("tts"))
+                trueorfalse = option.split("{tts:")[1].split("}")[0];
+
+            if (data && data.Return)
+                data.Return.tts = checkBoolean(trueorfalse, true);
+            else
+                Options.tts = checkBoolean(trueorfalse, true);
+          }
```
Comment (Faf4a) [inline src/handler/parsers.js:572]: Inefficient use of variables.

File: src/handler/parsers.js [hunk 5]
```diff
@@ -522,42 +49,778 @@ const SlashOptionsParser = async (options) => {
...
+            const trueorfalse = "true";
+
+            if (Checker("deleteCommand"))
+                trueorfalse = option.split("{deleteCommand:")[1].split("}")[0];
+
+            if (data.newMessageData && data.newMessageData.options)
+                data.newMessageData.options.deleteCommand = checkBoolean(trueorfalse, true);
+            else
+                Options.deleteCommand = checkBoolean(trueorfalse, true);
+          }
```
Comment (Faf4a) [inline src/handler/parsers.js:613]: Inefficient use of variables.

File: src/handler/parsers.js [hunk 6]
```diff
@@ -522,42 +49,778 @@ const SlashOptionsParser = async (options) => {
...
+            const trueorfalse = "true";
+
+            if (Checker("fetchReply"))
+                trueorfalse = option.split("{fetchReply:")[1].split("}")[0];
+
+            if (data && data.Return)
+                data.Return.fetchReply = checkBoolean(trueorfalse, true);
+            else
+                Options.fetchReply = checkBoolean(trueorfalse, true);
+          }
```
Comment (Faf4a) [inline src/handler/parsers.js:625]: Inefficient use of variables.

File: src/handler/parsers.js [hunk 7]
```diff
@@ -522,42 +49,778 @@ const SlashOptionsParser = async (options) => {
...
+            let Files = [];
+            
+            if (data.Checker(code, "file")) {
+                const files = code.split("{file:")?.slice(1)?.map((file) => file?.trim());
+
+                for (let file of files) {
+                    var params = file?.split("}")[0]?.split(":");
+
+                    console.log(params);
+
```
Comment (Faf4a) [inline src/handler/parsers.js:657]: Unneeded `console.log`.

Comment (devlordduck) [reply src/handler/parsers.js:657]: i just noticed

File: src/handler/parsers.js (hunk 2 above)
Comment (devlordduck) [reply src/handler/parsers.js:288]: ok

**Comment by USERSATOSHI** [comment]
can u fix the conflicts then we can be ready

**Comment by Leref** [comment]
Has not resolved conflicts, will be dismissed.

**Comment by devlordduck** [comment]
> can u fix the conflicts then we can be ready

fine

**Comment by devlordduck** [comment]
> Has not resolved conflicts, will be dismissed.

i dont get notifications from github

**Comment by devlordduck** [comment]
and i wasnt on github for some time

**Pull request closed without merging**